### PR TITLE
storage: media upload, raw http test, prep for batch endpoint

### PR DESCRIPTION
**Pull request by dragonsinth**
@mahmoudbahaa see these changes related to https://github.com/fullstorydev/emulators/pull/15

Wanna give your PR a spin on top of these new tests and see what happens?

### CODE REVIEW

File: storage/Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ vet:
 
 .PHONY: staticcheck
 staticcheck:
-	@go install honnef.co/go/tools/cmd/staticcheck@v0.3.3
+	@go install honnef.co/go/tools/cmd/staticcheck@v0.4.3
```
Comment (gpassini) [inline storage/Makefile:27]: Do we need this upgrade? It seems to have broken the build for older versions of Go:
```
# honnef.co/go/tools/go/loader
../../go/pkg/mod/honnef.co/go/tools@v0.4.3/go/loader/loader.go:233:4: unknown field 'Instances' in struct literal of type types.Info
note: module requires Go 1.19
```

File: storage/gcsemu/raw_http_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	// Create the object 1.
+	w := oh.NewWriter(ctx)
+	assert.NilError(t, write(w, v1))
+
+	// Make sure object 2 is not there.
+	_ = bh.Object(name2).Delete(ctx)
+
+	// Run each test individually.
+	for _, tc := range tcs {
+		tc := tc
```
Comment (gpassini) [inline storage/gcsemu/raw_http_test.go:151]: Nit: Do we need to shadow this variable? I thought `t.Run` blocks while the test is running, unless we explicitly specify `t.Parallel()` inside it, so these tests will run sequentially, right?

File: storage/Makefile (hunk 1 above)
Comment (dragonsinth) [reply storage/Makefile:27]: Oh dang.. the old version doesn't work on 1.20.  No worries, will revert.

File: storage/gcsemu/raw_http_test.go (hunk 2 above)
Comment (dragonsinth) [reply storage/gcsemu/raw_http_test.go:151]: True, this was kind of defensive coding.

File: storage/Makefile (hunk 1 above)
Comment (dragonsinth) [reply storage/Makefile:27]:
```suggestion
	@go install honnef.co/go/tools/cmd/staticcheck@v0.3.3
```

**Review by gpassini — APPROVED** [review]
LGTM!

**Pull request merged**